src/include/duckdb.h

```diff
@@ -1471,6 +1471,17 @@ Returns `DUCKDB_TYPE_INVALID` if the parameter index is out of range or the stat
 */
 DUCKDB_API duckdb_type duckdb_param_type(duckdb_prepared_statement prepared_statement, idx_t param_idx);
 
+/*!
+Returns the parameter logical type for the parameter at the given index.
+
+Returns `nullptr` if the parameter index is out of range or the statement was not successfully prepared.
```

Comment:
Can we specify that the created `duckdb_logical_type` has to be cleaned up and with what function that has to happen?
Looking at `duckdb_column_logical_type`:
```
The return type of this call should be destroyed with `duckdb_destroy_logical_type`.
```